sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/persistence/DatabasePersistenceAndroidTest.java

```diff
@@ -370,27 +369,26 @@ public void putTooManyLogs() throws PersistenceException {
         persistence.setLogSerializer(logSerializer);
         try {
 
-            /* Generate some logs that will be evicted. */
-            for (int i = 0; i < 20; i++) {
-                persistence.putLog(AndroidTestUtils.generateMockLog(), "test-p1", PERSISTENCE_NORMAL);
-            }
-
-            /*
-             * Generate the maximum number of logs that we can store in this configuration.
-             * This will evict all previously stored logs.
-             */
-            List<Log> expectedLogs = new ArrayList<>();
-            for (int i = 0; i < 10; i++) {
+            /* Generate logs until we notice eviction. */
+            List<Log> allLogs = new ArrayList<>();
+            String group = "test-p1";
+            for (int i = 0; allLogs.size() == persistence.countLogs(group); i++) {
                 MockLog log = AndroidTestUtils.generateMockLog();
-                persistence.putLog(log, "test-p1", PERSISTENCE_NORMAL);
-                expectedLogs.add(log);
+                persistence.putLog(log, group, PERSISTENCE_NORMAL);
+                allLogs.add(log);
+
+                /* Fail if no eviction happen after a long time to avoid infinite loop on bug. */
```

Comment:
"happen" => "happens", same for https://github.com/Microsoft/AppCenter-SDK-Android/pull/871/commits/aa06618b4c812c673b1cdf1bb60a6aace74475de#diff-13c7c0c5167ac2fcdb51679095067306R429